### PR TITLE
[KAYAKO-1459]:Unit test for KreCaseSubscription

### PR DESCRIPTION
**Pull request by ssankarpradhan**
https://jira.devfactory.com/browse/KAYAKO-1459
Before
image
After
image
Coverage gain : 164

### CODE REVIEW

**Comment by carloscaldas-aurea** [comment]
Your Pull Request has been selected for a project-agnostic code review by [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members).

<details><summary>More information</summary>

_This is an initiative to improve overall code quality. If your change is **urgent**, you **don't** have to wait for our review._ 

_NOTE: This review **does not** replace your team's review._

</details>

File: ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/kre/base/kase/KreCaseSubscriptionTest.java
Comment (carloscaldas-aurea) [inline ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/kre/base/kase/KreCaseSubscriptionTest.java]: ✅ Good to see constant used instead of hard-coded values.

This approach increases maintainability (single point of modification) and debugging, avoid duplication and decreases verbosity.

**No change suggested, looks good!**	  

<a id="a2c85680-4980-11e8-86d4-9358f2b1b645" href="none" codereview-uuid-reserved-tag />

File: ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/kre/base/kase/KreCaseSubscriptionTest.java
Comment (carloscaldas-aurea) [inline ProjectSource/app-library/src/test/java/com/kayako/sdk/android/k5/kre/base/kase/KreCaseSubscriptionTest.java]: ⚠️  Here, the code is using `Collection.size()` for testing emptiness. Although, using `Collection.size()` to test for emptiness works, using `Collection.isEmpty()` makes the code more readable and can be more efficient. The time complexity of any `isEmpty()` method implementation should be **O(1)** whereas some implementations of `size()` can be **O(n)**.  

**[More details](https://confluence.devfactory.com/x/uyCkEw)**

## The solution in this case

```JAVA
errorCollector.checkThat(mClientTypingListenersList.isEmpty(), is(true));
```
 

<a id="cf943bc0-4980-11e8-86d4-9358f2b1b645" href="none" codereview-uuid-reserved-tag />

**Review by carloscaldas-aurea — COMMENTED** [review]
Product-agnostic code review by [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members) completed. 

<table>
  <tr>
    <td>:no_entry_sign:<b>Severes:</b>  0</td>
    <td>:exclamation:<b>Reds:</b>  0</td>
    <td>:warning:️<b>Yellows:</b>  1</td>
    <td>:white_check_mark:<b>Greens:</b>  1</td>
  </tr>
</table>

<details><summary>More information</summary>
  <table>
    <tr>
      <td>:no_entry_sign:</td>
      <td>Severe Findings point out serious violations which can affect stability, performance, maintainability and therefore should be fixed.
        <ul>
          <li>Please note that the merge might have been blocked until all Severe Findings are fixed.</li>
          <li>While it is highly recommended to implement all findings only Severe Findings might have been blocking the merge</li>
        </ul>
      </td>
    </tr>
    <tr>
      <td>:exclamation:</td>
      <td>Red Findings represent potential bugs and it is highly recommended to implement them.<br/>
      Not required for the approval of the PR
      </td>
    </tr>
    <tr>
      <td>:warning:️</td>
      <td>Yellow Findings are suggestions to improve the code, warnings and tips<br/>
      Not required for the approval of the PR
      </td>
    </tr>
    <tr>
      <td>:white_check_mark:</td>
      <td>Green Findings highlight good ideas and solutions</td>
    </tr>
    <tr>
      <td>:repeat:</td>
      <td>Repeated instances of the same findings are counted just once and flagged with the repeat icon</td>
    </tr>
  </table>

  Feel free to reply to this comment, (make sure you `@tag` us) so we can start a conversation. The goal here is to learn and grow together and make our products better.

  > NOTE: This review **does not** replace your team's review._
</details>
<br/>
<b>How likely would you recommend the Code Review service to another developer?</b>

Not at all likely \[ [1](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/kayako-Kayako-Android-SDK/pull/247&entry.1163746017=1) \] \[ [2](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/kayako-Kayako-Android-SDK/pull/247&entry.1163746017=2) \] \[ [3](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/kayako-Kayako-Android-SDK/pull/247&entry.1163746017=3) \] \[ [4](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/kayako-Kayako-Android-SDK/pull/247&entry.1163746017=4) \] \[ [5](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/kayako-Kayako-Android-SDK/pull/247&entry.1163746017=5) \] \[ [6](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/kayako-Kayako-Android-SDK/pull/247&entry.1163746017=6) \] \[ [7](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/kayako-Kayako-Android-SDK/pull/247&entry.1163746017=7) \] \[ [8](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/kayako-Kayako-Android-SDK/pull/247&entry.1163746017=8) \] \[ [9](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/kayako-Kayako-Android-SDK/pull/247&entry.1163746017=9) \] \[ [10](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/kayako-Kayako-Android-SDK/pull/247&entry.1163746017=10) \] Very likely

> TIP: Press `Ctrl` (or `⌘` in a Mac) to open the survey in a new tab.

**Review by bartosz-rybusinski — APPROVED** [review]

**Pull request merged**
